boot/env.mli

```diff
@@ -102,6 +102,19 @@ val corelib : t -> Path.t
 (** coq/lib directory, not sure if to keep this *)
 val coqlib : t -> Path.t
 
+val ocamlfind : unit -> string
+
+val docdir : unit -> string
+
+val print_config : ?prefix_var_name:string -> t -> out_channel -> unit
+
+(** Needs a Rocq environment for Where and Config.
+    If the [usage] argument is [None], the query must not be PrintHelp. *)
+val print_query : Usage.specific_usage option -> Usage.query -> unit
+
+(** Where and Config need to be able to find coqlib (i.e. -boot won't work) *)
+val query_needs_env : Usage.query -> bool
```

Comment:
I think that in general the API design that would work best here is indeed having types `CoqEnv.t` and `OCamlEnv.t` if you want that _must_ be used by all the functions accessing values related to them.

This means that the upper layers must build values of these types, but this is IMHO what you want toplevels to do, and also, in order to ensure good user experience, you want to forbid ah-hoc calls.

Reply:
Not sure what the point of splitting env would be here, --config uses bits of both AFAIK

Reply:
Indeed I was thinking how to separate the cases where we need the prelude and where we don't.